src/controllers/v1/_common/demand.ts

```diff
@@ -82,15 +86,56 @@ export class DemandController extends Controller {
     if (updated_since) {
       query.push(['updated_at', '>', parseDateParam(updated_since)])
     }
+    const selfAddress = (await this.addressResolver.determineSelfAddress()).address
+    const roles = this.env.ROLES
+    const demands: DemandRow[] = []
+
+    if (roles.includes('optimiser') || roles.includes('admin')) {
+      // Privileged roles get all demands matching base query
+      demands.push(...(await this.db.get('demand', query)))
+      return await Promise.all(demands.map(async (demand) => responseWithAlias(demand, this.identity)))
+    }
+    // Owner's own demands
+    const ownedQuery: Where<'demand'> = [['owner', '=', selfAddress]]
+    if (updated_since) {
+      ownedQuery.push(['updated_at', '>', parseDateParam(updated_since)])
+    }
+    const ownedDemands = await this.db.get('demand', ownedQuery)
+
+    if (ownedDemands.length === 0) {
+      return await Promise.all(demands.map(async (demand) => responseWithAlias(demand, this.identity)))
+    }
+
+    demands.push(...ownedDemands.filter((demand) => demand.subtype === this.dbDemandSubtype))
 
-    const demands = await this.db.get('demand', query)
-    const result = await Promise.all(demands.map(async (demand) => responseWithAlias(demand, this.identity)))
-    return result
+    const ownedDemandsOppositeSubTypeIds = ownedDemands
+      .filter((demand) => demand.subtype !== this.dbDemandSubtype)
+      .map((demand) => demand.id)
+
+    // Get all demands that are matched with the users demands
+    const match2Query: Where<'match2'> = [
+      [this.demandType === 'demandA' ? 'demand_b_id' : 'demand_a_id', 'IN', ownedDemandsOppositeSubTypeIds],
+    ]
+    const matches = await this.db.get('match2', match2Query)
+    const notOwnedDemandIds = matches.map((m) => (this.dbDemandSubtype === 'demand_a' ? m.demand_a_id : m.demand_b_id))
+    if (notOwnedDemandIds.length > 0) {
+      const matchedQuery: Where<'demand'> = [...query, ['id', 'IN', notOwnedDemandIds]]
+      if (updated_since) {
+        matchedQuery.push(['updated_at', '>', parseDateParam(updated_since)])
+      }
+      demands.push(...(await this.db.get('demand', matchedQuery)))
+    }
+
+    return await Promise.all(demands.map(async (demand) => responseWithAlias(demand, this.identity)))
```

Comment:
`demands.map` could be something like `[...filteredOwnedDemands, ...matchedDemands].map`